maintenance/cloudwatch-event-parser/README.md

```diff
@@ -0,0 +1,130 @@
+* **Confirm changes before deploy**: If set to yes, any change sets will be shown to you before execution for manual review. If set to no, the AWS SAM CLI will automatically deploy application changes.
+* **Allow SAM CLI IAM role creation**: Many AWS SAM templates, including this example, create AWS IAM roles required for the AWS Lambda function(s) included to access AWS services. By default, these are scoped down to minimum required permissions. To deploy an AWS CloudFormation stack which creates or modified IAM roles, the `CAPABILITY_IAM` value for `capabilities` must be provided. If permission isn't provided through this prompt, to deploy this example you must explicitly pass `--capabilities CAPABILITY_IAM` to the `sam deploy` command.
+* **Save arguments to samconfig.toml**: If set to yes, your choices will be saved to a configuration file inside the project, so that in the future you can just re-run `sam deploy` without parameters to deploy changes to your application.
+
+## Useage
+This Lambda function is designed to be used in a CloudWatch event rule. Just 
+set the rule target to this Lambda function.
+
+## Use the SAM CLI to build and test locally
+
+Build your application 
+
+```bash
+cloudwatch-event-parser$ sam build
+```
+
+The SAM CLI installs dependencies defined in `json_formatter/requirements.txt`, creates a deployment package, and saves it in the `.aws-sam/build` folder.
+
+Test a single function by invoking it directly with a test event. An event is a JSON document that represents the input that the function receives from the event source. Test events are included in the `events` folder in this project.
+
+Run functions locally and invoke them with the `sam local invoke` command.
+
+```
+cloudwatch-event-parser$ sam local invoke JsonFormatterFunction --event events/event.json
+```
+Note: local invoke may not yield desired outcome due to inadequate permission 
+context support of SAM
+
+### Step-through debugging with VS Code
+
+#### Prerequisites
+All prerequisites listed in *Deploy the application* section above plus
+
+* [VS Code](https://code.visualstudio.com/)
+* (Optionally) [Python](https://marketplace.visualstudio.com/items?itemName=ms-python.python) extension for VS Code 
+
+To enable step-through debugging, 
+ 1. Open [app.code-workspace](./app.code-workspace) in VS Code
+ 2. uncomment following 3 lines in [json_formatter/app.py](./json_formatter/app.py#L7)
+
+    ```
+    # import ptvsd
+    # ptvsd.enable_attach(address=('0.0.0.0', 5890), redirect_output=True)
+    # ptvsd.wait_for_attach()
+    ```
+ 3. Set a breakpoint in json_formatter/app.py below the above uncommented lines
+ 4. Run
+
+    ```
+    sam build
+    sam local invoke -e events/event.json -d 5890 JsonFormatterFunction
+    ```
+    The last cmd will hung waiting for a debugger to be attached.
+ 5. Hit `F5` to start debugging
```

Comment:
At this step I received errors:
```
Unable to import 'ptvsd'
Unable to import 'boto3'
```
running from VSCode on MacOS with an added `/.vscode/launch.json` file based on the sample at https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-using-debugging-python.html as:
```
{
    "version": "0.2.0",
    "configurations": [
        {
           "name": "SAM CLI Python Hello World",
           "type": "python",
           "request": "attach",
           "port": 5890,
           "host": "localhost",
           
           "pathMappings": [
               {
                   "localRoot": "${workspaceFolder}",
                   "remoteRoot": "/var/task"
               }
           ]
       }
   ]
 }
```

the Python interpreter I'd set VSCode to was `/usr/local/bin/python3` which is "_Python3.7.7 64-bit_".

I noticed while running in the terminal inside VSCode that:
```
$ which python
/usr/local/bin/python
mwalle@gdx-mw-macbook ~/Documents/GitHub/GDX-Analytics/maintenance/cloudwatch-event-parser (json_formatter)
mwalle@gdx-mw-macbook ~/Documents/GitHub/GDX-Analytics/maintenance/cloudwatch-event-parser (json_formatter)
$ python --version
Python 2.7.17
```